src/modules/siptrace/siptrace.c

```diff
@@ -636,85 +690,93 @@ static int sip_trace_store_db(siptrace_data_t *sto)
 	return -1;
 }
 
-static int fixup_siptrace(void **param, int param_no)
+enum siptrace_type_t parse_siptrace_flag(str* sflags)
 {
-	char *duri;
-	sip_uri_t uri;
-	dest_info_t *dst = NULL;
-	proxy_l_t *p = NULL;
-	str dup_uri_str = {0, 0};
+	int idx;
+	enum siptrace_type_t trace_type = SIPTRACE_NONE;
 
-	if(param_no != 1) {
-		LM_DBG("params:%s\n", (char *)*param);
-		return 0;
+	if (sflags == NULL || sflags->s == NULL || sflags->len == 0) {
+		return SIPTRACE_NONE;
 	}
 
-	if(((char *)(*param))[0] == '\0') {
-		// Empty URI, use the URI set at module level (dup_uri)
-		if(dup_uri) {
-			uri = *dup_uri;
-		} else {
-			LM_ERR("Missing duplicate URI\n");
-			return -1;
-		}
-	} else {
-		duri = (char *)*param;
+	for (idx = 0; idx < sflags->len; idx++) {
```

Comment:
This looks that there a a lot of whitespaces, can you check? Maybe github shows it different.

Reply:
it's github diff that's behaving odd.